modules/catalog-compaction/src/integrationTest/java/org/apache/ignite/internal/catalog/compaction/ItCatalogCompactionTest.java

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    @Test
+    void testGlobalMinimumTxBeginTime() {
+        IgniteImpl node0 = CLUSTER.node(0);
+        IgniteImpl node1 = CLUSTER.node(1);
+        IgniteImpl node2 = CLUSTER.node(2);
+
+        List<CatalogCompactionRunner> compactors = List.of(
+                node0.catalogCompactionRunner(),
+                node1.catalogCompactionRunner(),
+                node2.catalogCompactionRunner()
+        );
+
+        Collection<ClusterNode> topologyNodes = node0.clusterNodes();
+
+        InternalTransaction tx1 = (InternalTransaction) node0.transactions().begin();
+        InternalTransaction tx2 = (InternalTransaction) node1.transactions().begin();
+        InternalTransaction readonlyTx = (InternalTransaction) node1.transactions().begin(new TransactionOptions().readOnly(true));
+        InternalTransaction tx3 = (InternalTransaction) node2.transactions().begin();
+
+        compactors.forEach(compactor -> {
+            TimeHolder timeHolder = await(compactor.determineGlobalMinimumRequiredTime(topologyNodes, 0L));
+            assertThat(timeHolder.minActiveTxBeginTime, is(tx1.startTimestamp().longValue()));
+        });
+
+        tx1.rollback();
+
+        compactors.forEach(compactor -> {
+            TimeHolder timeHolder = await(compactor.determineGlobalMinimumRequiredTime(topologyNodes, 0L));
+            assertThat(timeHolder.minActiveTxBeginTime, is(tx2.startTimestamp().longValue()));
+        });
+
+        tx2.commit();
+
+        compactors.forEach(compactor -> {
+            TimeHolder timeHolder = await(compactor.determineGlobalMinimumRequiredTime(topologyNodes, 0L));
+            assertThat(timeHolder.minActiveTxBeginTime, is(tx3.startTimestamp().longValue()));
+        });
+
+        tx3.rollback();
+
+        // Since there are no active RW transactions in the cluster, the minimum time will be min(now()) across all nodes.
+        compactors.forEach(compactor -> {
+            long minTime = Stream.of(node0, node1, node2).map(node -> node.clockService().nowLong()).min(Long::compareTo).orElseThrow();
+
+            TimeHolder timeHolder = await(compactor.determineGlobalMinimumRequiredTime(topologyNodes, 0L));
+
+            long maxTime = Stream.of(node0, node1, node2).map(node -> node.clockService().nowLong()).min(Long::compareTo).orElseThrow();
+
+            // Read-only transactions are not counted,
+            assertThat(timeHolder.minActiveTxBeginTime, greaterThan(readonlyTx.startTimestamp().longValue()));
+
+            assertThat(timeHolder.minActiveTxBeginTime, greaterThanOrEqualTo(minTime));
+            assertThat(timeHolder.minActiveTxBeginTime, lessThanOrEqualTo(maxTime));
+        });
+
+        readonlyTx.rollback();
+    }
+
+    private static void ensureTimestampStoredInAllReplicas(HybridTimestamp expTime, int expTablesCount) throws InterruptedException {
+        Int2IntMap tablesWithPartitions = catalogManagerFacade().collectTablesWithPartitionsBetween(
```

Comment:
Generally it's not a good idea to test something that uses method X with method X itself. I mean that 
`propagateTimeToReplicas` uses `collectTablesWithPartitionsBetween`.

Reply:
Thanks, removed this dependency